src/etos_lib/kubernetes/provider.py

```diff
@@ -0,0 +1,33 @@
+# Copyright Axis Communications AB.
+#
+# For a full list of individual contributors, please see the commit history.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Provider custom resource manager ETOS."""
+from .etos import Kubernetes, Resource
+
+
+class Provider(Resource):
+    """Provider handles the Provider custom Kubernetes resources."""
+
+    def __init__(self, client: Kubernetes, strict: bool = False):
+        """Set up Kubernetes client.
+
+        :param strict: If True, the client will raise exceptions when Kubernetes could not
+        be reached as expected such as the ETOS namespace not being able to be determined.
+        The default (False) will just ignore any problems.
+        """
+        self.strict = strict
```

Comment:
When will strict=False be used? Maybe log exceptions anyway even in non-strict mode?

Reply:
strict=False is used when we are doing Kubernetes operations but don't yet know if we are running in the controller or not and don't want to fail. For instance
```
if testrun.exists():
    # Do stuff within controller context
else:
    # To stuff outside controller context
```

I don't think we need logging in those cases since we are actively ignoring

Reply:
Given docstring and the explanation above I still don't understand when false is to be used.